apps/worker/src/app/workflow/usecases/send-message/send-message-voice.usecase.ts

```diff
@@ -0,0 +1,365 @@
+          detail: DetailEnum.SUBSCRIBER_NO_CHANNEL_DETAILS,
+          source: ExecutionDetailsSourceEnum.INTERNAL,
+          status: ExecutionDetailsStatusEnum.FAILED,
+          isTest: false,
+          isRetry: false,
+        }),
+        command.organizationId
+      );
+
+      return;
+    }
+    if (!integration) {
+      await this.sendErrorStatus(
+        message,
+        'warning',
+        'voice_missing_integration_error',
+        'Subscriber does not have an active voice integration',
+        command,
+        LogCodeEnum.MISSING_VOICE_INTEGRATION
+      );
+
+      const metadata = CreateExecutionDetailsCommand.getExecutionLogMetadata();
+      await this.executionLogQueueService.add(
+        metadata._id,
+        CreateExecutionDetailsCommand.create({
+          ...metadata,
+          ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
+          messageId: message._id,
+          detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION,
+          source: ExecutionDetailsSourceEnum.INTERNAL,
+          status: ExecutionDetailsStatusEnum.FAILED,
+          isTest: false,
+          isRetry: false,
+        }),
+        command.organizationId
+      );
+
+      return;
+    }
+    if (!integration?.credentials?.from) {
+      await this.sendErrorStatus(
+        message,
+        'warning',
+        'no_integration_from_phone',
+        'Integration does not have from phone configured',
+        command,
+        LogCodeEnum.MISSING_VOICE_PROVIDER
+      );
+      const metadata = CreateExecutionDetailsCommand.getExecutionLogMetadata();
+      await this.executionLogQueueService.add(
+        metadata._id,
+        CreateExecutionDetailsCommand.create({
+          ...metadata,
+          ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
+          messageId: message._id,
+          detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_CHANNEL,
+          source: ExecutionDetailsSourceEnum.INTERNAL,
+          status: ExecutionDetailsStatusEnum.FAILED,
+          isTest: false,
+          isRetry: false,
+        }),
+        command.organizationId
+      );
+
+      return;
+    }
+  }
+
+  private async sendMessage(
+    phone: string,
+    integration: IntegrationEntity,
+    content: string,
+    message: MessageEntity,
+    command: SendMessageCommand,
+    overrides: Record<string, any> = {}
+  ) {
+    try {
+      const voiceFactory = new VoiceFactory();
+      const voiceHandler = voiceFactory.getHandler(this.buildFactoryIntegration(integration));
+      if (!voiceHandler) {
+        throw new PlatformException(`Voice handler for provider ${integration.providerId} is  not found`);
+      }
+
+      const result = await voiceHandler.send({
+        to: overrides.to || phone,
+        from: overrides.from || integration.credentials.from,
+        content: overrides.content || content,
+        record: overrides.record,
+        language: overrides.language,
+        voice: overrides.voice,
```

Comment:
should these come from templates eventually? or subscriber credentials?
is this fine for now?